components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -1099,18 +1099,44 @@ fun stringOrNull(jsonObject: JSONObject, key: String): String? {
     }
 }
 
+enum class SearchResultReason {
+    KEYWORD,
+    ORIGIN,
+    URL,
+    PREVIOUS_USE,
+    BOOKMARK,
+    TAG;
+
+    companion object {
+        private val values = values()
+        fun fromMessage(reason: MsgTypes.SearchResultReason): SearchResultReason {
+            return when (reason) {
+                MsgTypes.SearchResultReason.KEYWORD -> KEYWORD
+                MsgTypes.SearchResultReason.ORIGIN -> ORIGIN
+                MsgTypes.SearchResultReason.URL -> URL
+                MsgTypes.SearchResultReason.PREVIOUS_USE -> PREVIOUS_USE
+                MsgTypes.SearchResultReason.BOOKMARK -> BOOKMARK
+                MsgTypes.SearchResultReason.TAG -> TAG
+            }
+        }
+    }
+}
+
 data class SearchResult(
     val url: String,
     val title: String,
-    val frecency: Long
-    // Skipping `reasons` for now...
+    val frecency: Long,
+    val reasons: List<SearchResultReason>
 ) {
     companion object {
         internal fun fromMessage(msg: MsgTypes.SearchResultMessage): SearchResult {
             return SearchResult(
                 url = msg.url,
                 title = msg.title,
-                frecency = msg.frecency
+                frecency = msg.frecency,
+                reasons = msg.reasonsList.map {
+                    SearchResultReason.fromMessage(it)
+                }.toTypedArray().toList()
```

Comment:
Nit: Was there a way to get a list without the double conversion?

Reply:
Ah sorry, this looks silly...`map` returns `List` already.